apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/StandardResponseAdapters.kt

```diff
@@ -0,0 +1,93 @@
+package com.apollographql.apollo.api.internal
+
+import com.apollographql.apollo.api.internal.json.JsonReader
+import com.apollographql.apollo.api.internal.json.JsonWriter
+
+class ListResponseAdapter<T>(private val wrappedAdapter: ResponseAdapter<T>): ResponseAdapter<List<T>> {
+  override fun fromResponse(reader: JsonReader): List<T> {
+    reader.beginArray()
+    val list = mutableListOf<T>()
+    while(reader.hasNext()) {
+      list.add(wrappedAdapter.fromResponse(reader))
+    }
+    reader.endArray()
+    return list
+  }
+
+  override fun toResponse(writer: JsonWriter, value: List<T>) {
+    writer.beginArray()
+    value.forEach {
+      wrappedAdapter.toResponse(writer, it)
+    }
+    writer.endArray()
+  }
+}
+
+class NullableResponseAdapter<T:Any>(private val wrappedAdapter: ResponseAdapter<T>): ResponseAdapter<T?> {
+  override fun fromResponse(reader: JsonReader): T? {
+    return if (reader.peek() == JsonReader.Token.NULL) {
+      reader.skipValue()
+      null
+    } else {
+      wrappedAdapter.fromResponse(reader)
+    }
+  }
+
+  override fun toResponse(writer: JsonWriter, value: T?) {
+    if (value == null) {
+      writer.nullValue()
+    } else {
+      wrappedAdapter.toResponse(writer, value)
+    }
+  }
+}
+
+object stringResponseAdapter: ResponseAdapter<String> {
```

Comment:
why lower case?

Reply:
No specific reason. Reading: https://kotlinlang.org/docs/coding-conventions.html#property-names, both upper and lower case seem to work.

Reply:
IntelliJ says that should be uppercase too. I made everything uppercase